jobs/gcp-discovery/src/main/java/com/tmobile/pacbot/gcp/inventory/collector/ServiceAccountInventoryCollector.java

```diff
@@ -40,10 +42,36 @@ public  List<ServiceAccountVH> fetchServiceAccountDetails(ProjectVH project) thr
            serviceAccountVH.setDescription(serviceAccount.getDescription());
            logger.info("Serice account inventory collector-> {}",serviceAccount.getName());
            getServiceAccountKeys(serviceAccount.getUniqueId(),serviceAccountVH,project);
+           serviceAccountMap.put(serviceAccount.getName(), serviceAccountVH);
+
+           //Method to fetch the role and membership of accounts of project
+           fetchMembership(serviceAccountVH,project.getProjectId());
 
            serviceAccountVHList.add(serviceAccountVH);
+
        }
-    return serviceAccountVHList;
+       return serviceAccountVHList;
+    }
+    private void fetchMembership(ServiceAccountVH serviceAccountVH, String projectId){
+        CloudResourceManager.Projects.GetIamPolicy iamPolicy = null;
+        try {
+            Map<String,List<String>> rolesMembers=new HashMap<>();
+            iamPolicy = gcpCredentialsProvider.getCloudResourceManager().projects().getIamPolicy(projectId, new GetIamPolicyRequest());
+            List<com.google.api.services.cloudresourcemanager.model.Binding> binds = iamPolicy.execute().getBindings();
+            if(binds!=null) {
+                for (com.google.api.services.cloudresourcemanager.model.Binding binding : binds) {
+                    logger.info("Binding role: {}",binding.getRole());
+                    logger.info("Binding members: {}",binding.getMembers());
+                    rolesMembers.put(binding.getRole(),binding.getMembers());
+                }
+            }
+            serviceAccountVH.setRolesMembers(rolesMembers);
```

Comment:
Service account should have list of only those roles, of which its a memberOf.
In this case its having role-membership of every account.